### ISSUE TITLE
Re-investigate download speed regression with 0.16.0-betaX

### ISSUE THREAD

**Issue by wjt** [endlessm/kolibri-explore-plugin#736]
> _Originally posted by @vanessa-chang in https://github.com/endlessm/kolibri-explore-plugin/issues/729#issuecomment-1669560694_
> Test on a chromebook, the download time takes longer compared to the production page.
> ![chromebook](https://github.com/endlessm/kolibri-explore-plugin/assets/31236978/00fcb97c-1a3e-430d-8c80-a88f8c2dfeb7)

Quoth @dbnicholson:

> It would be nice to verify that downloads are actually fixed, but I think this is for the best and any further regressions should probably be pursued upstream.

As I recall the original regression was more like an order of magnitude, so downloads "only" taking twice as long is still an improvement.

@dbnicholson wrote a script to benchmark different Kolibri versions:

https://gist.github.com/dbnicholson/c9ffc907546d8b866ad6210f6442a2e1

It would be interesting to rerun this with 0.16.0-betaX

**Comment by dbnicholson** [comment]
Perusing some logs flying by on Android I still see all the partial range requests (with 206 responses). I can't really make sense of the [`FileDownload` class](https://github.com/learningequality/kolibri/blob/234c3de02820cfa52ed62bfbebd3eb85893802c0/kolibri/utils/file_transfer.py#L495) anymore, but it's certainly never going to be as fast as a single request for the whole file. What I don't get is that it seems like the incremental download case is already supported out of the box using requests' streaming support. Write the content to the file as it's received and update the progress according to the written size vs the size in the `Content-Length` header. I must be missing something, though.

What I think would be nice but would certainly require upstream agreement would be specifying whether a full or incremental download is desired. In the collection downloader I almost certainly think we just want to get the files as fast as possible.

**Comment by dbnicholson** [comment]
I don't think we have the resources to keep digging into this right now, so yes I think it will have to go in the backlog. My hope is to fix #592 this sprint and just do less downloading to get out of the welcome screen faster.
